File: Lab/Lab01/Recu.java
package Lab01;

import java.util.Scanner;

public class Recu {
    // metodo que chama total de maiusculos
    public static int total(String palavra) {
        // chamar 2 função
        return total(palavra, 0);
    }

    // metodo recursivo que realmente faz o teste
    public static int total(String palavra, int x) {
        // definir dados
        int cont = 0;
        char letra = 'a';
        // ponto de parada
        if (x < palavra.length()) {
            letra = palavra.charAt(x);
            if (Character.isUpperCase(letra)) {
                cont++;
            }
            cont += total(palavra, x + 1);
        }
        // retornar
        return cont;
    }

    // função main
    public static void main(String[] args) {
        // definir dados
        String palavra = "";
        int totM = 0;
        Scanner scanner = new Scanner(System.in);
        // fazer metódo enquanto a palavra 'FIM' não for encontrada
        do {
            palavra = scanner.nextLine();
            if (!palavra.equals("FIM")) {
                totM = total(palavra);
                System.out.println("O total de maiusculas é " + totM);
            }
        } while (!palavra.equals("FIM"));
        // lembrar de fechar scanner
        scanner.close();
    }
}

File: Lab/Lab01/Recu.c
#include <stdio.h>
#include <string.h>

// função de recursão
int totM(char *palavra, int x)
{
    // definir dados
    char letra = 'a';
    int cont = 0;
    int tam = strlen(palavra);
    // ponto de parada
    if (x < tam)
    {
        // testar se o caractere daquela pos é M
        letra = palavra[x];
        if ('A' <= letra && letra <= 'Z')
        {
            cont = cont + 1;
        }
        // recursão
        cont += totM(palavra, x + 1);
    }
    // retornar valor
    return (cont);
}

// função de chamada
int tot(char *palavra)
{
    // retornar x
    return (totM(palavra, 0));
}

// função main
int main()
{
    // definir dados
    char palavra[80];
    int total = 0;
    // ler palavra e chamar função enquanto diferente de FIM
    do
    {
        scanf("%s", palavra);
        getchar();
        if (strcmp(palavra, "FIM"))
        {
            total = tot(palavra);
            printf("O total de M e %i\n", total);
        }
    } while (strcmp(palavra, "FIM"));
    // retornar
    return 0;
}
